mfa/commons/src/main/kotlin/com/pingidentity/mfa/commons/MfaConfiguration.kt

```diff
@@ -21,7 +21,7 @@ import com.pingidentity.logger.Logger
  *           for security reasons, as an attacker could potentially access cached credentials from memory dumps.
  * @property logger The logger instance used for logging messages. Defaults to a global logger instance.
  */
-class MfaConfiguration private constructor(
+open class MfaConfiguration(
```

Comment:
May able to simplify as below, so PushConfiguration and OathConfiguration can just extend it and add its own configuration attribute.:

```
open class MfaConfiguration {
    var encryptionEnabled: Boolean = true
    var timeoutMs: Long = DEFAULT_TIMEOUT_MS
    var enableCredentialCache: Boolean = false
    var logger: Logger = Logger.logger
    val context: Context
        get() = ContextProvider.context
        
    companion object {
        // Timeout for the HTTP client, default is 15 seconds
        const val DEFAULT_TIMEOUT_MS = 15000L
        operator fun invoke(block: MfaConfiguration.() -> Unit = {}): MfaConfiguration {
            val config = MfaConfiguration()
            config.apply(block) // apply the configuration block
            return config
        }
    }
}
```

Reply:
Applied the suggested approach for all configs.